src/GraphQL.NewtonsoftJson/ExecutionResultJsonConverter.cs

```diff
@@ -109,6 +109,6 @@ private void WriteExtensions(ExecutionResult result, JsonWriter writer, JsonSeri
 
         public override bool CanRead => false;
 
-        public override bool CanConvert(Type objectType) => objectType == typeof(ExecutionResult);
+        public override bool CanConvert(Type objectType) => typeof(ExecutionResult).IsAssignableFrom(objectType);
```

Comment:
This change is not mandatory and does not relate to the described problem, I just decided that it would be better to do it symmetrically xplicitly.